From c187c23028df93345fa283d7cf98a698c423bd15 Mon Sep 17 00:00:00 2001
From: Andrew Aye <github.very069@passmail.net>
Date: Sat, 30 Aug 2025 00:01:56 -0700
Subject: [PATCH 5/5] Implement deferred DLL function loading.

---
 src/prim/windows/prim.c | 19 +++++++++++++------
 1 file changed, 13 insertions(+), 6 deletions(-)

diff --git a/src/prim/windows/prim.c b/src/prim/windows/prim.c
index 1531183e..4f4845d4 100644
--- a/src/prim/windows/prim.c
+++ b/src/prim/windows/prim.c
@@ -106,26 +106,26 @@ static bool win_enable_large_os_pages(size_t* large_page_size)
   // <https://devblogs.microsoft.com/oldnewthing/20110128-00/?p=11643>
   unsigned long err = 0;
   HANDLE token = NULL;
-  BOOL ok = OpenProcessToken(GetCurrentProcess(), TOKEN_ADJUST_PRIVILEGES | TOKEN_QUERY, &token);
+  BOOL ok = pOpenProcessToken(GetCurrentProcess(), TOKEN_ADJUST_PRIVILEGES | TOKEN_QUERY, &token);
   if (ok) {
     TOKEN_PRIVILEGES tp;
-    ok = LookupPrivilegeValue(NULL, TEXT("SeLockMemoryPrivilege"), &tp.Privileges[0].Luid);
+    ok = pLookupPrivilegeValueA(NULL, TEXT("SeLockMemoryPrivilege"), &tp.Privileges[0].Luid);
     if (ok) {
       tp.PrivilegeCount = 1;
       tp.Privileges[0].Attributes = SE_PRIVILEGE_ENABLED;
-      ok = AdjustTokenPrivileges(token, FALSE, &tp, 0, (PTOKEN_PRIVILEGES)NULL, 0);
+      ok = pAdjustTokenPrivileges(token, FALSE, &tp, 0, (PTOKEN_PRIVILEGES)NULL, 0);
       if (ok) {
-        err = GetLastError();
+        err = pGetLastError();
         ok = (err == ERROR_SUCCESS);
         if (ok && large_page_size != NULL && pGetLargePageMinimum != NULL) {
           *large_page_size = (*pGetLargePageMinimum)();
         }
       }
     }
-    CloseHandle(token);
+    pCloseHandle(token);
   }
   if (!ok) {
-    if (err == 0) err = GetLastError();
+    if (err == 0) err = pGetLastError();
     _mi_warning_message("cannot enable large OS page support, error %lu\n", err);
   }
   return (ok!=0);
@@ -187,6 +187,13 @@ void _mi_prim_mem_init( mi_os_mem_config_t* config )
         }
       }
     }
+    pCloseHandle = (PCloseHandle)(void (*)(void))GetProcAddress(hDll, "CloseHandle");
+    pGetLastError = (PGetLastError)(void (*)(void))GetProcAddress(hDll, "GetLastError");
+    pGetCurrentProcess = (PGetCurrentProcess)(void (*)(void))GetProcAddress(hDll, "GetCurrentProcess");
+    pOpenProcessToken = (POpenProcessToken)(void (*)(void))GetProcAddress(hDll, "OpenProcessToken");
+    pLookupPrivilegeValueA = (PLookupPrivilegeValueA)(void (*)(void))GetProcAddress(hDll, "LookupPrivilegeValueA");
+    pAdjustTokenPrivileges = (PAdjustTokenPrivileges)(void (*)(void))GetProcAddress(hDll, "AdjustTokenPrivileges");
+    pGetLargePageMinimum = (PGetLargePageMinimum)(void (*)(void))GetProcAddress(hDll, "GetLargePageMinimum");
     FreeLibrary(hDll);
   }
   // Enable large/huge OS page support?
-- 
2.47.1.windows.1

