From 97c88a545cd7fcd6f1b7677ddb5bf361e3a480af Mon Sep 17 00:00:00 2001
From: Andrew Aye <github.very069@passmail.net>
Date: Fri, 15 Nov 2024 20:40:35 -0800
Subject: [PATCH] Stufsdfa

---
 common_defs.h | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/common_defs.h b/common_defs.h
index 094d438..891ecb6 100644
--- a/common_defs.h
+++ b/common_defs.h
@@ -30,13 +30,9 @@
 
 #include "libdeflate.h"
 
-#include <stdbool.h>
-#include <stddef.h>	/* for size_t */
-#include <stdint.h>
 #ifdef _MSC_VER
-#  include <intrin.h>	/* for _BitScan*() and other intrinsics */
-#  include <stdlib.h>	/* for _byteswap_*() */
    /* Disable MSVC warnings that are expected. */
+#  pragma warning(disable : 4391) /* incorrect return type for intrinsic function, expected 'unsigned char' */
    /* /W2 */
 #  pragma warning(disable : 4146) /* unary minus on unsigned type */
    /* /W3 */
@@ -51,7 +47,12 @@
 #  pragma warning(disable : 4232) /* nonstandard extension used */
 #  pragma warning(disable : 4245) /* conversion from 'int' to 'unsigned int' */
 #  pragma warning(disable : 4295) /* array too small to include terminating null */
+#  include <intrin.h>	/* for _BitScan*() and other intrinsics */
+#  include <stdlib.h>	/* for _byteswap_*() */
 #endif
+#include <stdbool.h>
+#include <stddef.h>	/* for size_t */
+#include <stdint.h>
 #ifndef FREESTANDING
 #  include <string.h>	/* for memcpy() */
 #endif
-- 
2.46.2.windows.1

